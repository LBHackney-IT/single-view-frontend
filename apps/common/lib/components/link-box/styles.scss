.mtfh-link-overlay a {
  position: static;

  &:before {
    content: "";
    cursor: inherit;
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 0;
    width: 100%;
    height: 100%;
  }
}

.mtfh-link-box {
  position: relative;

  a[href]:not(.mtfh-link-box a),
  abbr[title] {
    position: relative;
    z-index: 1;
  }
}
