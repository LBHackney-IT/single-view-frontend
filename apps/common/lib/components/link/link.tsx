import React, { forwardRef } from "react";

import cn from "classnames";

import { widthOverrides } from "../../utils";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface LinkProps {
  variant?: "link" | "danger" | "text-colour" | "muted" | "back-link" | "native";
  isExternal?: boolean;
  override?: number;
}

export type LinkComponent = Polymorphic.ForwardRefComponent<"a", LinkProps>;

export const Link: LinkComponent = forwardRef(function Link(
  {
    as: LinkComp = "a",
    variant = "link",
    isExternal = false,
    className,
    rel,
    target,
    override,
    ...props
  },
  ref,
) {
  const linkClasses = cn(
    variant !== "native" && {
      "govuk-link lbh-link": variant !== "back-link",
      "govuk-back-link lbh-back-link": variant === "back-link",
      [`lbh-link--${variant}`]: variant !== "link" && variant !== "back-link",
      "lbh-link--no-visited-state": !isExternal,
    },
    widthOverrides(override),
    className,
  );
  return (
    // eslint-disable-next-line react/jsx-no-target-blank
    <LinkComp
      ref={ref}
      className={linkClasses}
      rel={isExternal ? "noopener noreferrer" : rel}
      target={isExternal ? "_blank" : target}
      {...props}
    />
  );
});
