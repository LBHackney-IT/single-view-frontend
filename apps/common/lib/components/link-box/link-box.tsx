import React, { ReactElement, forwardRef } from "react";

import cn from "classnames";

import { widthOverrides } from "../../utils";
import { LinkProps } from "../link";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface LinkOverlayProps {
  children: ReactElement<LinkProps>;
  override?: number;
}

export type LinkOverlayComponent = Polymorphic.ForwardRefComponent<
  "div",
  LinkOverlayProps
>;

export const LinkOverlay: LinkOverlayComponent = forwardRef(function LinkOverlay(
  { as: LinkOverlayComp = "div", className, override, ...props },
  ref,
) {
  return (
    <LinkOverlayComp
      ref={ref}
      className={cn("mtfh-link-overlay", widthOverrides(override), className)}
      {...props}
    />
  );
});

export interface LinkBoxProps {
  override?: number;
}

export type LinkBoxComponent = Polymorphic.ForwardRefComponent<"div", LinkBoxProps>;

export const LinkBox: LinkBoxComponent = forwardRef(function LinkBox(
  { as: LinkBoxComp = "div", className, override, ...props },
  ref,
) {
  return (
    <LinkBoxComp
      ref={ref}
      className={cn("mtfh-link-box", widthOverrides(override), className)}
      {...props}
    />
  );
});
