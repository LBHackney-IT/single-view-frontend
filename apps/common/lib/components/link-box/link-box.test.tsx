import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Link } from "../link";
import { LinkBox, LinkOverlay } from "./link-box";

test("it renders correctly", async () => {
  const { container } = render(
    <LinkBox>
      <LinkOverlay>
        <Link href="https://localhost">Link</Link>
      </LinkOverlay>
    </LinkBox>,
  );

  expect(container).toMatchSnapshot();
  await testA11y(container);
});
