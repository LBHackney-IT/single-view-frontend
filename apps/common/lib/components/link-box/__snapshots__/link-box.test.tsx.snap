// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="mtfh-link-box"
  >
    <div
      class="mtfh-link-overlay"
    >
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="https://localhost"
      >
        Link
      </a>
    </div>
  </div>
</div>
`;
